tests/integration/start_test.go

```diff
@@ -0,0 +1,263 @@
+	})
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+	require.NotEmpty(t, resp.TaskId)
+
+	var statusResp *pb.TaskStatusResponse
+
+	require.Eventually(t, func() bool {
+		statusResp, err = client.GetTaskStatus(ctx, &pb.TaskStatusRequest{
+			TaskId: resp.TaskId,
+		})
+		// wait for the task to exit with and error
+		return err == nil && statusResp != nil && statusResp.Status == pb.JobStatus_JOB_STATUS_EXITED_ERROR
+	}, 3*time.Second, pollInterval, "expected task to exit with error")
+
+	assert.Equal(t, pb.JobStatus_JOB_STATUS_EXITED_ERROR, statusResp.Status)
+	assert.Equal(t, "", statusResp.TerminationSource)
+	assert.Equal(t, "", statusResp.TerminationSignal)
+	assert.NotNil(t, statusResp.EndTime)
+	assert.NotNil(t, statusResp.StartTime)
+	assert.Equal(t, resp.TaskId, statusResp.TaskId)
+	assert.Equal(t, exitCode2, *statusResp.ExitCode)
+}
+
+func TestIntegration_StartTaskCommandDoesNotExistInPath(t *testing.T) {
+	t.Parallel()
+
+	client := createTestClient(t, "client001")
+
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+
+	resp, err := client.StartTask(ctx, &pb.StartTaskRequest{
+		Command: "test-command-that-does-not-exist",
+	})
+	require.Error(t, err)
+	require.Nil(t, resp)
+	sts, ok := status.FromError(err)
+	require.True(t, ok)
+	assert.Equal(t, codes.InvalidArgument, sts.Code())
+	assert.Contains(t, sts.Message(), "invalid command")
+	assert.Contains(t, sts.Message(), "executable file not found in $PATH")
+}
+
+func TestIntegration_StartTaskFullPathCommandDoesNotExist(t *testing.T) {
+	t.Parallel()
+
+	client := createTestClient(t, "client001")
+
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+
+	resp, err := client.StartTask(ctx, &pb.StartTaskRequest{
+		Command: "/path/to/test-command-that-does-not-exist",
+	})
+	require.Error(t, err)
+	require.Nil(t, resp)
+	sts, ok := status.FromError(err)
+	require.True(t, ok)
+	assert.Equal(t, codes.InvalidArgument, sts.Code())
+	assert.Contains(t, sts.Message(), "command not found")
+	assert.Contains(t, sts.Message(), "no such file or directory")
+}
+
+func TestIntegration_StartTask(t *testing.T) {
+	t.Parallel()
+
+	client := createTestClient(t, "client001")
+
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+
+	resp, err := client.StartTask(ctx, &pb.StartTaskRequest{
+		Command: "ls",
+		Args:    []string{"-l"},
+	})
+	assert.NotNil(t, resp)
+	assert.NotEmpty(t, resp.TaskId)
+	require.NoError(t, err)
+	sts, ok := status.FromError(err)
+	require.True(t, ok)
+	assert.Equal(t, codes.OK, sts.Code())
+}
+
+func TestIntegration_StartTaskWithFullPath(t *testing.T) {
+	t.Parallel()
+
+	client := createTestClient(t, "client001")
+
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+
+	resp, err := client.StartTask(ctx, &pb.StartTaskRequest{
+		Command: "/bin/ls",
+		Args:    []string{"-l"},
+	})
+	assert.NotNil(t, resp)
```

Comment:
```suggestion
	require.NotNil(t, resp)
```

same issue here, this will panic if the previous call fails; non exhaustive, there's several assertions like this